stdlib/public/core/Arrays.swift.gyb

```diff
@@ -2181,37 +2162,81 @@ extension ${Self} {
 %end
 
 #if _runtime(_ObjC)
+import SwiftShims
+
+internal enum _ArrayBridgeFailure : Error {
+case count(Int)
+}
+
+extension _ContiguousArrayBuffer {
+  internal init?(_bridgingFromObjectiveC source: AnyObject) {
+    typealias Storage = _ContiguousArrayStorage<Element>
+    
+    // First see if there's already a native buffer we can grab
+    if let storage = (source as? Storage)
+      ?? (source as? _SwiftDeferredNSArray)?._nativeStorage as? Storage {
+      self.init(storage)
+    }
+    else {
+      self.init(_slowBridgingFromObjectiveC: source)
+    }
+  }
+
+  internal init?(_slowBridgingFromObjectiveC source: AnyObject) {
+    let nsArray = unsafeBitCast(source, to: _NSArrayCore.self)
+    let count = nsArray.count
+    self.init(_uninitializedCount: count)
+    if count == 0 { return }
+
+    if (_isBridgedVerbatimToObjectiveC(Element.self)) {
+      withUnsafeMutableBufferPointer {
+        $0.baseAddress!.withMemoryRebound(to: AnyObject.self, capacity: count) {
+          nsArray.getObjects($0, range: _SwiftNSRange(location: 0, length: count))
```

Comment:
FWIW, I believe that Swift follows ARC in treating `Foo **` as `Foo * __autoreleasing *` (i.e. AUMP), but because of the existence of MRR code will never actually optimize based on that fact. However, that doesn't mean the API is suddenly safe, because nothing is owning the objects while they're in the buffer.

We didn't want to use Unmanaged for this because that implies that whether to treat these as +0 or +1 is context-dependent, which it isn't. __unsafe_unretained on out-parameters always means +0.